camel-idea-plugin/src/main/java/org/apache/camel/idea/service/extension/camel/XmlCamelIdeaUtils.java

```diff
@@ -31,21 +47,119 @@
 
     @Override
     public boolean isCamelRouteStart(PsiElement element) {
-        if (element.getText().equals("from") || element.getText().equals("rest")) {
+        if (element instanceof XmlTag) {
+            return isCamelRouteStartTag((XmlTag) element);
+        } else if (element.getText().equals("from") || element.getText().equals("rest")) {
             XmlTag xml = PsiTreeUtil.getParentOfType(element, XmlTag.class);
-            if (xml != null) {
-                String name = xml.getLocalName();
-                XmlTag parentTag = xml.getParentTag();
-                if (parentTag != null) {
-                    boolean xmlEndTag = element.getPrevSibling().getText().equals("</");
-                    return "routes".equals(parentTag.getLocalName()) && "rest".equals(name) && !xmlEndTag
-                            || "route".equals(parentTag.getLocalName()) && "from".equals(name);
-                }
+            boolean xmlEndTag = element.getPrevSibling().getText().equals("</");
+            if (xml != null && !xmlEndTag) {
+                return isCamelRouteStartTag(xml);
             }
         }
         return false;
     }
 
+    @Override
+    public boolean isCamelRouteStartExpression(PsiElement element) {
+        if (element instanceof XmlToken
+            && !element.getText().equals("<")
+            && !element.getText().equals("</")
+            && !element.getText().equals(">")
+            && !element.getText().equals("/>")) {
+            return isCamelRouteStart(element);
+        } else {
```

Comment:
No need for the else branch since you return early in the if branch

Just at the end of the method return false